src/utilities.ts

```diff
@@ -450,112 +450,112 @@ export async function computeShortAggregatorAddress(
+                "uint256"     // thresholdPrice - Indication of the price the user is willing to pay (e.g. ethers.utils.parseUnits(priceWillingToPay, await usdc_e.decimals()).toString()).
+            ],
+            [
+                optionOrderParams.orderType === OrderType.LONG_OPEN ||  optionOrderParams.orderType == OrderType.SHORT_OPEN,
+                optionOrderParams.ticker,
+                unixExpiration,
+                optionOrderParams.expiration,
+                bigNumberStrike,
+                formattedStrike,
+                optionOrderParams.contractType,
+                intQuantity,
+                thresholdPrice
+            ]
+        )
+        // Note that we are signing a message, not a transaction
+        const signature = await wallet.signMessage(
+            ethers.utils.arrayify(hashedValues)
+        )
+        const value = optionOrderParams.thresholdPrice! * optionOrderParams.quantity!
+        let amountToApprove: ethers.BigNumber
+
+        if(optionOrderParams.orderType === OrderType.SHORT_OPEN) {
+            let diffPrice: number = 0;
+            if (optionOrderParams.contractType == 1 || optionOrderParams.contractType == 0){
+                // put
+                diffPrice = Number(optionOrderParams.strike[0])
+            }
+            else if (optionOrderParams.contractType == 2){
+                // call spread
+                diffPrice = Math.abs(Number(optionOrderParams.strike[1]) - Number(optionOrderParams.strike[0]))
+            }
+            else if (optionOrderParams.contractType == 3){
+                // put spread
+                diffPrice = Math.abs(Number(optionOrderParams.strike[0]) - Number(optionOrderParams.strike[1]))
+            }
+            amountToApprove = ethers.utils.parseUnits(
+                (
+                    optionOrderParams.quantity! * diffPrice).toString(),
+                    stablecoinDecimals
+                )
+        } else {
+            amountToApprove = ethers.BigNumber.from
             (
-                optionOrderParams.quantity! * diffPrice).toString(),
-                stablecoinDecimals
+                ethers.utils.parseUnits(value.toFixed(stablecoinDecimals), stablecoinDecimals)
             )
-    } else {
-        amountToApprove = ethers.BigNumber.from
-        (
-            ethers.utils.parseUnits(value.toFixed(stablecoinDecimals), stablecoinDecimals)
-        )
-    }
+        }
    
-    return {
-        hashedValues,
-        signature,
-        amountToApprove,
-        ...optionOrderParams,
-        unixExpiration,
-        formattedStrike,
-        bigNumberStrike,
-        bigNumberThresholdPrice: thresholdPrice
-    }
+        return {
+            hashedValues,
+            signature,
+            amountToApprove,
+            ...optionOrderParams,
+            unixExpiration,
+            formattedStrike,
+            bigNumberStrike,
+            bigNumberThresholdPrice: thresholdPrice
+        }
+    }));
```

Comment:
It might make sense to have the internal function here be separated out so that you can have `const preparedParams = await Promise.all(optionOrderParamsList.map(externalFunctionForReadability))`

Reply:
But this is not a requirement for this PR